sdk/src/main/java/io/opentelemetry/sdk/trace/export/SpanData.java

```diff
@@ -0,0 +1,544 @@
+
+  /**
+   * The id of this span, retrieved from the associated SpanContext.
+   *
+   * @return the SpanId
+   */
+  public SpanId getSpanId() {
+    return getContext().getSpanId();
+  }
+
+  /**
+   * An immutable implementation of {@link io.opentelemetry.trace.Link}.
+   *
+   * @since 0.1.0
+   */
+  @Immutable
+  @AutoValue
+  public abstract static class Link implements io.opentelemetry.trace.Link {
+    private static final Map<String, AttributeValue> EMPTY_ATTRIBUTES = Collections.emptyMap();
+
+    /**
+     * Returns a new {@code Link}.
+     *
+     * @param context the context of the linked {@code Span}.
+     * @return a new {@code Link}.
+     * @since 0.1.0
+     */
+    public static Link create(SpanContext context) {
+      return new AutoValue_SpanData_Link(context, EMPTY_ATTRIBUTES);
+    }
+
+    /**
+     * Returns a new {@code Link}.
+     *
+     * @param context the context of the linked {@code Span}.
+     * @param attributes the attributes of the {@code Link}.
+     * @return a new {@code Link}.
+     * @since 0.1.0
+     */
+    public static Link create(SpanContext context, Map<String, AttributeValue> attributes) {
+      return new AutoValue_SpanData_Link(
+          context, Collections.unmodifiableMap(new HashMap<>(attributes)));
+    }
+
+    Link() {}
+  }
+
+  /**
+   * An immutable implementation of the {@link io.opentelemetry.trace.Event}.
+   *
+   * @since 0.1.0
+   */
+  @Immutable
+  @AutoValue
+  public abstract static class Event implements io.opentelemetry.trace.Event {
+    private static final Map<String, AttributeValue> EMPTY_ATTRIBUTES =
+        Collections.unmodifiableMap(Collections.<String, AttributeValue>emptyMap());
+
+    /**
+     * Returns a new {@code Event} with the given name.
+     *
+     * @param name the text name of the {@code Event}.
+     * @return a new {@code Event} with the given name.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @since 0.1.0
+     */
+    public static io.opentelemetry.trace.Event create(String name) {
+      return new AutoValue_SpanData_Event(name, EMPTY_ATTRIBUTES);
+    }
+
+    /**
+     * Returns a new {@code Event} with the given name and set of attributes.
+     *
+     * @param name the text name of the {@code Event}.
+     * @param attributes the attributes of the {@code Event}.
+     * @return a new {@code Event} with the given name and set of attributes.
+     * @throws NullPointerException if {@code name} or {@code attributes} are {@code null}.
+     * @since 0.1.0
+     */
+    public static io.opentelemetry.trace.Event create(
+        String name, Map<String, AttributeValue> attributes) {
+      return new AutoValue_SpanData_Event(
+          name,
+          Collections.unmodifiableMap(new HashMap<>(Utils.checkNotNull(attributes, "attributes"))));
+    }
+  }
+
+  /**
+   * A timed event representation.
+   *
+   * @since 0.1.0
+   */
+  @Immutable
+  @AutoValue
+  public abstract static class TimedEvent {
```

Comment:
Change TimedEvent to:
* timestamp
* name
* attributes

And remove the Event from the SpanData. A TimedEvent can be constructed from:
* timestamp + api.Event
* timestamp + name
* timestamp + name + attributes

Reply:
done

Reply:
we're adding javadoc now